src/Control/Middleware/ConfirmationMiddleware/HttpMethodBypass.php

```diff
@@ -46,17 +46,9 @@ public function getMethods()
     public function addMethods(...$methods)
     {
         // uppercase and exclude empties
-        $methods = array_reduce(
-            $methods,
-            static function &(&$result, $method) {
-                $method = strtoupper(trim($method));
-                if (strlen($method)) {
-                    $result[] = $method;
-                }
-                return $result;
-            },
-            []
-        );
+        $methods = array_filter(array_map(function ($method) {
```

Comment:
`array_reduce` is better here, IMO as it is one iteration rather than two over the array

Reply:
Ah, that explains the original construction. In that case it needs to be rewritten to avoid the pass-by-reference pieces, I believe. This was failing in PHP8.

Reply:
I believe it should keep working if we just remove ampersands there. The original pass-by-reference been initially done to optimise for PHP 5.6. In PHP 7 they've made tricky internal optimisations for arrays (no extra memcpy anymore), so I guess that'll be working in almost the same way without passing by reference.

Reply:
Reverted, dropped ampersands, working fine 😄